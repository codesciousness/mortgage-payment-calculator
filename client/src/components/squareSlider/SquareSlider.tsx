import React from 'react';
import Slider from '@mui/material/Slider';

type SquareSliderProps = {
    name: string;
    defaultValue: number;
    min: number;
    max: number;
    steps: number;
}

const SquareSlider = ({ name, defaultValue, min, max, steps }: SquareSliderProps): JSX.Element => {

    return (
        <div>
            <label htmlFor={name} style={{ display: 'block' }}>{name}</label>
            <Slider
                id={name}
                name={name}
                aria-label={name}
                defaultValue={defaultValue}
                valueLabelDisplay="on"
                min={min}
                max={max}
                step={steps}
                sx={{
                    width: 300,
                    margin: '2rem 1rem',
                    '& .MuiSlider-thumb': {
                        borderRadius: '1px',
                        color: 'DeepSkyBlue'
                    },
                    '& .MuiSlider-track': {
                        background: 'linear-gradient(to right, PaleTurquoise, PowderBlue, SkyBlue, DeepSkyBlue)',
                        border: 'none'
                    },
                    '& .MuiSlider-rail': {
                        color: 'LightBlue'
                    }
                }}
            />
        </div>
    );
}

export default SquareSlider;