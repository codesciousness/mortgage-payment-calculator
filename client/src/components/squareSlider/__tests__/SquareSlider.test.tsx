import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import SquareSlider from '../SquareSlider';

const name = 'Label Name';
const value = 25;
const min = 0;
const max = 50;
const steps = 1;
const width = 300;
const onChange = () => null;

describe('<SquareSlider />', () => {
    test('SquareSlider component is defined', () => {
        expect(<SquareSlider name={name} value={value} min={min} max={max} 
            steps={steps} width={width} onChange={onChange}/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<SquareSlider name={name} value={value} min={min} max={max} 
            steps={steps} width={width} onChange={onChange}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly with props', () => {
        render(<SquareSlider name={name} value={value} min={min} max={max} 
            steps={steps} width={width} onChange={onChange}/>);
        expect(screen.getByText('Label Name')).toBeInTheDocument();
        expect(screen.getByDisplayValue('25')).toBeInTheDocument();
    });
});