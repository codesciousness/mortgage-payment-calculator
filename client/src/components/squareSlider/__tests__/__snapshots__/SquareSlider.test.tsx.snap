// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SquareSlider /> component matches the snapshot 1`] = `
<div
  className="SquareSlider"
  style={
    Object {
      "marginBottom": "0.75rem",
    }
  }
>
  <label
    className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-22ecsy-MuiFormLabel-root-MuiInputLabel-root"
    id="LabelName"
  >
    Label Name
  </label>
  <span
    className="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-1pa4zyv-MuiSlider-root"
    id="LabelName"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
    />
    <span
      className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "50%",
        }
      }
    />
    <span
      className="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
      data-index={0}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "left": "50%",
          "pointerEvents": undefined,
        }
      }
    >
      <input
        aria-label="Label Name"
        aria-orientation="horizontal"
        aria-valuemax={50}
        aria-valuemin={0}
        aria-valuenow={25}
        data-index={0}
        disabled={false}
        max={50}
        min={0}
        name="Label Name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        step={1}
        style={
          Object {
            "border": 0,
            "clip": "rect(0 0 0 0)",
            "direction": "ltr",
            "height": "100%",
            "margin": -1,
            "overflow": "hidden",
            "padding": 0,
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "100%",
          }
        }
        type="range"
        value={25}
      />
      <span
        aria-hidden={true}
        className="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-17djbj3-MuiSlider-valueLabel"
      >
        <span
          className="MuiSlider-valueLabelCircle"
        >
          <span
            className="MuiSlider-valueLabelLabel"
          >
            25
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;
